library/x509_crt.c

```diff
@@ -590,6 +590,74 @@ static int x509_get_ext_key_usage( unsigned char **p,
     return( 0 );
 }
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108 at this point.
+ */
+static int x509_get_other_name( unsigned char **p,
+                                const unsigned char *end,
+                                mbedtls_x509_name *hardware_module_name )
+{
+    int ret;
+    size_t len;
+    unsigned char tag;
+    mbedtls_x509_buf cur_oid = { 0, 0, NULL };
+
+     if( ( ret = mbedtls_asn1_get_tag( p, end, &len,
+                                       MBEDTLS_ASN1_OID ) ) != 0 )
+         return( MBEDTLS_ERR_X509_INVALID_EXTENSIONS + ret );
+
+     cur_oid.tag = MBEDTLS_ASN1_OID;
+     cur_oid.p = *p;
+     cur_oid.len = len;
+
+     /*
+      * Skip other otherName SAN
+      */
+     if( MBEDTLS_OID_CMP( MBEDTLS_OID_ON_HW_MODULE_NAME, &cur_oid ) != 0 )
+     {
+         *p += len;
+         return( 0 );
+     }
+     *p += len;
+    if( ( ret = mbedtls_asn1_get_tag( p, end, &len,
```

Comment:
Minor: The indentation is off here.

Reply:
yes, the indentation error described in https://github.com/ARMmbed/mbedtls/pull/2530#discussion_r268996754 is continued until this line